packages/opentelemetry-instrumentation-http/src/utils.ts

```diff
@@ -177,6 +177,66 @@ export const setSpanWithError = (
   span.setStatus(status);
 };
 
+/**
+ * Adds attributes for request content-length and content-encoding HTTP headers
+ * @param { IncomingMessage } Request object whose headers will be analyzed
+ * @param { Attributes } Attributes object to be modified
+ */
+export const setRequestContentLengthAttribute = (
+  request: IncomingMessage,
+  attributes: Attributes
+) => {
+  const length = getContentLength(request.headers);
+  if (length === null) return;
+
+  if (isCompressed(request.headers)) {
+    attributes[HttpAttribute.HTTP_REQUEST_CONTENT_LENGTH] = length;
+  } else {
+    attributes[HttpAttribute.HTTP_REQUEST_CONTENT_LENGTH_UNCOMPRESSED] = length;
+  }
+};
+
+/**
+ * Adds attributes for response content-length and content-encoding HTTP headers
+ * @param { IncomingMessage } Response object whose headers will be analyzed
+ * @param { Attributes } Attributes object to be modified
+ */
+export const setResponseContentLengthAttribute = (
+  response: IncomingMessage,
+  attributes: Attributes
+) => {
+  const length = getContentLength(response.headers);
+  if (length === null) return;
+
+  if (isCompressed(response.headers)) {
+    attributes[HttpAttribute.HTTP_RESPONSE_CONTENT_LENGTH] = length;
+  } else {
+    attributes[
+      HttpAttribute.HTTP_RESPONSE_CONTENT_LENGTH_UNCOMPRESSED
+    ] = length;
+  }
+};
+
+function getContentLength(
+  headers: OutgoingHttpHeaders | IncomingHttpHeaders
+): number | null {
+  const contentLengthHeader = headers['content-length'];
```

Comment:
At least for outgoing HTTP users may specify headers with any casing.

Reply:
They should be lower-cased by node when set: https://github.com/nodejs/node/blob/master/lib/_http_outgoing.js#L572

Reply:
You refer to the object used by node http internally on symbol `kOutHeaders` accessible via the apis `setHeader`/ `getHeader`/... on outgoing requests.
But actually this is not applicable here. outgoing messages have not .header property at all.

But it seems this is no problem because this plugin captures content-size only for server requests (incoming) and client response (also incoming) where node parses and prepares the lowercased headers.